table/block_based_table_reader.cc

```diff
@@ -1380,6 +1439,28 @@ class BlockBasedTable::BlockEntryIteratorState : public TwoLevelIteratorState {
   bool skip_filters_;
 };
 
+BlockBasedTable::IndexPartitionIteratorState::IndexPartitionIteratorState(
```

Comment:
Great catch! Due to similarity of index and data blocks the impl of IndexPartitionIteratorState seems to have evolved to be identical to that of BlockEntryIteratorState. I will remove it then.